src/main/java/seedu/command/AddCommand.java

```diff
@@ -0,0 +1,226 @@
+
+            updatedFields.add(fieldName);
+            updatedFieldTypes.put(fieldName, type);
+        }
+
+        inventory.setFields(updatedFields);
+        inventory.setFieldTypes(updatedFieldTypes);
+
+        csv.updateCsvHeaders(inventory);
+        ui.showFieldsAndRecords(inventory);
+    }
+
+    private void handleAddMultipleFields(String fieldData) throws InventraException {
+        if (fieldData.isEmpty()) {
+            throw new InventraMissingFieldsException();
+        }
+
+        String[] newFields = fieldData.split(",\\s*");
+        boolean success = true;
+
+        for (String field : newFields) {
+            String[] parts = field.split("/");
+            if (parts.length != 2) {
+                throw new InventraInvalidTypeException("Field format", field, "correct format (type/fieldName)");
+            }
+
+            String type = parts[0].trim();
+            String fieldName = parts[1].trim();
+
+            if (!isValidFieldType(type)) {
+                throw new InventraInvalidTypeException(fieldName, type, "valid field type (e.g., 's', 'i', 'f')");
+            }
+
+            if (inventory.getFields().contains(fieldName)) {
+                ui.printMessage("Field '" + fieldName + "' already exists. Cannot add duplicate headers.");
+                success = false;
+                continue;
+            }
+
+            inventory.addField(fieldName, type);
+        }
+
+        if (success) {
+            ui.showSuccessFieldsAdded();
+        } else {
+            ui.printMessage("Failed to add one or more fields due to errors.");
+        }
+
+        ui.showFieldsAndRecords(inventory);
+    }
+
+    private void handleAddRecord(String recordData) throws InventraException {
+        LOGGER.info("Handling add record: " + recordData);
+
+        if (inventory.getFields().isEmpty()) {
+            throw new InventraMissingFieldsException();
+        }
+
+        String[] values = recordData.split(",\\s*");
+
+        if (values.length != inventory.getFields().size()) {
+            throw new InventraInvalidRecordCountException(inventory.getFields().size(), values.length);
+        }
+
+        Map<String, String> record = new HashMap<>();
+        for (int i = 0; i < values.length; i++) {
+            String field = inventory.getFields().get(i);
+            String type = inventory.getFieldTypes().get(field);
+
+            assert type != null : "Type for field '" + field + "' should not be null";
+
+            String value = values[i].trim();
+
+            String validationMessage = validateValue(value, type, field);
+            if (validationMessage != null) {
+                ui.showValidationError(validationMessage);
+                return;
+            }
+
+            record.put(field, value);
+        }
+
+        inventory.addRecord(record);
+        ui.showSuccessRecordAdded();
+    }
+
+    private String validateValue(String value, String type, String field) throws InventraException {
```

Comment:
The method name validateValue did not hint to me that it will be returning a String.
Consider renaming the method / break down the method into smaller parts.